src/Cli/dotnet/UILanguageOverride.cs

```diff
@@ -24,7 +26,20 @@ public static void Setup()
 
         private static void ApplyOverrideToCurrentProcess(CultureInfo language)
         {
+
+            CultureInfo.DefaultThreadCurrentCulture = language;
             CultureInfo.DefaultThreadCurrentUICulture = language;
+
+            // The CurrentUICulture can be supposedly be left incorrectly unchanged even when DefaultThreadCurrentUICulture is changed:
+            // https://learn.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentculture?redirectedfrom=MSDN&view=net-7.0#remarks
+            CultureInfo.CurrentUICulture = language;
+
+            if (OperatingSystem.IsWindows() && Environment.OSVersion.Version.Major >= 10) // Encoding is only an issue on Windows, UTF-8 is only officially supported on 10+.
+            {
+                Console.OutputEncoding = DefaultMultilingualEncoding;
+                Console.InputEncoding = DefaultMultilingualEncoding; // Setting both encodings causes a change in the CHCP, making it so we dont need to P-Invoke ourselves.
```

Comment:
@adamsitnik would setting UTF-8 encoding to the console be ok?

Reply:
According to this, https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page, after the May 2019 Windows 10 update UTF-8 is officially supported. Since this only happens in Windows 10+, and when using languages besides English, which already have a bad encoding without this change, I think we can take this change for 8. 

cc @dsplaisted  for potential approval.